src/broker/messaging.bif

```diff
@@ -66,8 +66,16 @@ static bool publish_event_args(ArgsSpan args, const zeek::String* topic,
                                         "Did you mean to use Cluster::publish() instead of Broker::publish()?");
 
 	if ( args[0]->GetType()->Tag() == zeek::TYPE_RECORD )
-		rval = zeek::broker_mgr->PublishEvent(topic->CheckString(),
-		                                      args[0]->AsRecordVal());
+		{
+		auto* rv = args[0]->AsRecordVal();
+
+		// same_type() should be fast if it's the type pointers are the same.
+		if ( zeek::same_type(rv->GetType(), zeek::BifType::Record::Broker::Event) )
+			rval = zeek::broker_mgr->PublishEvent(topic->CheckString(), rv);
+		else
+			zeek::emit_builtin_error(zeek::util::fmt("expected Broker::Event, got %s",
+			                                         zeek::obj_desc_short(rv->GetType()).c_str()));
```

Comment:
Nit: suggest validation from normal flow. Also, the comment misses a.

```suggestion
		// Validate that the passed value type matches. `same_type()`
		// should be fast since it compares type pointers.
		if ( !zeek::same_type(rv->GetType(), zeek::BifType::Record::Broker::Event) )
			{
			zeek::emit_builtin_error(
				zeek::util::fmt(
					"expected Broker::Event, got %s",
					zeek::obj_desc_short(rv->GetType()).c_str()));

			return false;
			}

		rval = zeek::broker_mgr->PublishEvent(topic->CheckString(), rv);
```